src/Controller/ArticlesController.php

```diff
@@ -377,25 +378,47 @@ private function createFirstPage(Request $request, string $id, array $arguments)
                 }
 
                 if ($item->getReviewers()->notEmpty()) {
-                    $infoSections[] = ArticleSection::basic(
-                        'Reviewing Editor',
-                        3,
-                        $this->render(
-                            Listing::ordered(
-                                $item->getReviewers()
-                                    ->map(function (Reviewer $reviewer) {
-                                        $parts = [$reviewer->getPreferredName(), $reviewer->getRole()];
-
-                                        foreach ($reviewer->getAffiliations() as $affiliation) {
-                                            $parts[] = $affiliation->toString();
-                                        }
-
-                                        return implode(', ', $parts);
-                                    })
-                                    ->toArray()
-                            )
-                        )
-                    );
+                    $roles = $item->getReviewers()
+                        ->reduce(function (array $roles, Reviewer $reviewer) {
+                            $entry = $reviewer->getPreferredName();
+
+                            foreach ($reviewer->getAffiliations() as $affiliation) {
+                                $entry .= ", {$affiliation->toString()}";
+                            }
+
+                            $roles[$reviewer->getRole()][] = $entry;
+
+                            return $roles;
+                        }, []);
+
+                    uksort($roles, function (string $a, string $b) : int {
```

Comment:
Related: https://secure.php.net/manual/en/array.sorting.php says none of the functions in there are not necessarily `stable`; there may be a problem for multiple roles like `Reviewer`?

Reply:
Ah the roles are only appearing once, because they are keys; the existing order should be preserved by the `reduce`.